content/docs/context.md

```diff
@@ -4,39 +4,83 @@ title: Context
+```js
+<Provider value={/* some value */}>
+```
+
+A React component that allows Consumers to subscribe to context changes.
+
+Takes one prop, `value`, which will be passed to the [render prop](/docs/render-props.html) of child Consumers for the matching context anywhere in the component tree. One Provider can be connected to many Consumers.
+
+### `Consumer`
+
+```js
+<Consumer>
+ { value => { /* render something based on the context value */ } }
+</Consumer>
+```
 
-In some cases, you want to pass data through the component tree without having to pass the props down manually at every level.
-You can do this directly in React with the powerful "context" API.
+A React component that subscribes to context changes. 
+
+Takes a function as the `children` prop that receives the `value` prop of the matching Provider. This function will be called whenever the Provider's value is updated.
 
 > Note:
 >
-> A [new, safe version of context](https://github.com/reactjs/rfcs/blob/master/text/0002-new-version-of-context.md) is under development for the upcoming 16.3 release.
+> For more information about this pattern, see [render props](/docs/render-props.html).
+
+## Examples
+
+### Static Context
 
+Here is an example illustrating how you might inject a "theme" using context:
 
-## Why Not To Use Context
+`embed:context/theme-example.js`
 
-The vast majority of applications do not need to use context.
+### Dynamic Context
 
-If you want your application to be stable, don't use context. It is an experimental API and it is likely to break in future releases of React.
+A more complex example with dynamic values for the theme:
 
-If you aren't familiar with state management libraries like [Redux](https://github.com/reactjs/redux) or [MobX](https://github.com/mobxjs/mobx), don't use context. For many practical applications, these libraries and their React bindings are a good choice for managing state that is relevant to many components. It is far more likely that Redux is the right solution to your problem than that context is the right solution.
+**theme-context.js**
+`embed:context/theme-detailed-theme-context.js`
 
-If you're still learning React, don't use context. There is usually a better way to implement functionality just using props and state.
+**themed-button.js**
+`embed:context/theme-detailed-themed-button.js`
 
-If you insist on using context despite these warnings, try to isolate your use of context to a small area and avoid using the context API directly when possible so that it's easier to upgrade when the API changes.
+**app.js**
+`embed:context/theme-detailed-app.js`
 
-## How To Use Context
+## Motivation
 
-Suppose you have a structure like:
+Context is designed to relieve the pain of passing props down through a deeply nested component tree. For example, in the code below we manually thread through a color prop in order to style the Button and Message components. Using context, we can avoid passing props through intermediate elements.
```

Comment:
I'm still not a fan of this sequencing. We show motivation _after_ we show usage examples. That feels backwards to me. I think it would be better to present a problem, and then how to solve it using Context. Step by step.